pkg/test/fuzz/krmgen.go

```diff
@@ -112,3 +116,86 @@ func (rf *RandomFiller) fillWithRandom(t *testing.T, fieldName string, field ref
 		t.Fatalf("Unhandled field kind: %v", field.Kind())
 	}
 }
+
+type ZeroFiller struct {
+	fieldOverrides map[string]OverrideFiller
+}
+
+func NewZeroFiller(fc *FillerConfig) *ZeroFiller {
+	return &ZeroFiller{
+		fieldOverrides: fc.FieldOverrides,
+	}
+}
+
+func (rf *ZeroFiller) Fill(t *testing.T, obj interface{}) {
+	rf.fillWithZero(t, "", reflect.ValueOf(obj).Elem())
+}
+
+func (rf *ZeroFiller) fillWithZero(t *testing.T, fieldName string, field reflect.Value) {
+	if field.Kind() == reflect.Ptr {
+		if field.IsNil() {
+			field.Set(reflect.New(field.Type().Elem()))
+		}
+		rf.fillWithZero(t, fieldName, field.Elem())
+		return
+	}
+
+	if rf.fieldOverrides != nil {
+		if override, ok := rf.fieldOverrides[fieldName]; ok {
+			override(t, fieldName, field)
+			return
+		}
+	}
+
+	switch field.Kind() {
+	case reflect.Bool:
+		field.SetBool(false)
+
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		field.SetInt(0)
+
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		field.SetUint(0)
+
+	case reflect.Float32, reflect.Float64:
+		field.SetFloat(0.0)
+
+	case reflect.String:
+		field.SetString("")
+
+	case reflect.Slice: // undefined for now
+
+		// count := rf.randStream.Intn(10) + 1
```

Comment:
Should we silently skip slice and map fields, or should we explicitly fail? Alternatively, could we create empty slices and maps to use as the "zero value"?

Reply:
We can revisit this decision but I think ignoring more complex types for now is okay. As I wire this into the dev/ generation tools we will iterate here.